hbase-unsafe/src/main/java/org/apache/hadoop/hbase/unsafe/HBaseUnsafe0.java

```diff
@@ -0,0 +1,414 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.unsafe;
+
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Unsafe;
+
+/**
+ * Delegate all the method in sun.misc.Unsafe.
+ */
+@SuppressWarnings("restriction")
+final class HBaseUnsafe0 {
```

Comment:
This naming is unusual for our code base. We usually do things like HBaseUnsafe -> HBaseUnsafeImpl, or HBaseUnsafe -> HBaseUnsafeInternal. The '0' suffix is efficient but the more typical constructions are better for comprehension.

Reply:
I prefer the internal one.

Let me update the patch.